pkg/controller/storagecluster/storagecluster.go

```diff
@@ -317,6 +335,19 @@ func (c *Controller) syncStorageCluster(
 		return c.deleteStorageCluster(cluster)
 	}
 
+	// If portworx has been installed by daemonset, before it's uninstalled, operator
+	// should not proceed with installation.
+	if exists, err := c.isDaemonSetPresent(portworxDaemonSetName); err != nil || exists {
```

Comment:
Let's put this with other validations in validate() function (line 219).

Reply:
the problem with putting it in Validate() is, the operator uninstallation code will not be hit. when both DaemonSet and Operator are present, we should still let operator continue with uninstallation.